src/Og.php

```diff
@@ -267,26 +252,28 @@ public static function membershipDefault() {
     return ['type' => 'og_membership_type_default'];
   }
 
+
   /**
-   * Get all the modules fields that can be assigned to fieldable entities.
+   * Get an OG field base definition.
    *
-   * @param $field_name
-   *   The field name that was registered for the definition.
+   * @param string $plugin_id
+   *   The plugin ID, which is also the default field name.
    *
+   * @throws \Exception
    * @return OgFieldBase|bool
-   *   An array with the field and instance definitions, or FALSE if not.
-   *
-   * todo: pass the entity type and entity bundle to plugin definition.
+   *   An array with the field storage config and field config definitions, or
+   *   FALSE if none found.
    */
-  protected static function fieldInfo($field_name = NULL) {
-    $config = \Drupal::service('plugin.manager.og.fields');
-    $fields_config = $config->getDefinitions();
+  protected static function getFieldBaseDefinition($plugin_id) {
+    /** @var OgFieldsPluginManager $plugin_manager */
+    $plugin_manager = \Drupal::service('plugin.manager.og.fields');
```

Comment:
As mentioned already in chat yesterday. We will want to split this up
further anyway I think. So I would day let's leave real dependency
injection until the dust has settled with our api.
On 27 Nov 2015 14:03, "Pieter Frenssen" <notifications@github.com> wrote:

> In src/Og.php
> <https://github.com/amitaibu/og/pull/20#discussion_r46049162>:
>
> >     */
> > -  protected static function fieldInfo($field_name = NULL) {
> > -    $config = \Drupal::service('plugin.manager.og.fields');
> > -    $fields_config = $config->getDefinitions();
> > +  protected static function getFieldBaseDefinition($plugin_id) {
> > +    /** @var OgFieldsPluginManager $plugin_manager */
> > +    $plugin_manager = \Drupal::service('plugin.manager.og.fields');
>
> We should start injecting services instead of pulling them from the
> dependency injection container. I'll make a followup issue for this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/20/files#r46049162>.
>